keras_rs/src/layers/retrieval/brute_force_retrieval.py

```diff
@@ -0,0 +1,172 @@
+from typing import Any, Optional, Text
+
+import keras
+
+from keras_rs.src import types
+from keras_rs.src.api_export import keras_rs_export
+
+
+@keras_rs_export("keras_rs.layers.BruteForceRetrieval")
+class BruteForceRetrieval(keras.layers.Layer):
+    """Brute force top-k retrieval.
+
+    This layer maintains a set of candidates and is able to exactly retrieve the
+    top-k candidates for a given query. It does this by computing the scores for
+    all of the candidates for the query and extracting the top ones. The
+    returned top-k candidates are sorted by score.
+
+    By default, this layer returns a tuple with the top scores and top
+    identifiers, but it can be configured to return a single tensor with the top
+    identifiers.
+
+    The identifiers for the candidates can be specified as a tensor. If not
+    provided, the IDs used are simply the candidate indices.
+    """
+
+    def __init__(
+        self,
+        candidate_embeddings: Optional[types.Tensor] = None,
+        candidate_ids: Optional[types.Tensor] = None,
+        k: int = 10,
+        return_scores: bool = True,
+        name: Optional[Text] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initializes the layer.
+
+        Args:
```

Comment:
This section should be in the class docstring.

Reply:
Moved up.